tests/unit/contentproviders/test_hydroshare.py

```diff
@@ -0,0 +1,212 @@
+import json
+import os
+import pytest
+
+from contextlib import contextmanager
+from io import BytesIO
+from tempfile import TemporaryDirectory, NamedTemporaryFile
+from unittest.mock import patch
+from urllib.request import urlopen, Request, urlretrieve
+from zipfile import ZipFile
+
+from repo2docker.contentproviders import Hydroshare
+from repo2docker.contentproviders.base import ContentProviderException
+
+
+def test_content_id():
+    with patch.object(Hydroshare, "urlopen") as fake_urlopen:
+        fake_urlopen.return_value.url = (
+            "https://www.hydroshare.org/resource/b8f6eae9d89241cf8b5904033460af61"
+        )
+
+        def read():
+            return '{"dates": [{"type": "modified", "start_date": "2019-09-25T16:09:17.006152Z"}]}'
+
+        fake_urlopen.return_value.read = read
+        hydro = Hydroshare()
+
+        hydro.detect("10.4211/hs.b8f6eae9d89241cf8b5904033460af61")
+        assert hydro.content_id == "b8f6eae9d89241cf8b5904033460af61.v1569427757"
+
+
+test_hosts = [
+    (
+        [
+            "https://www.hydroshare.org/resource/b8f6eae9d89241cf8b5904033460af61",
+            "10.4211/hs.b8f6eae9d89241cf8b5904033460af61",
+            "https://doi.org/10.4211/hs.b8f6eae9d89241cf8b5904033460af61",
+        ],
+        {
+            "host": {
+                "hostname": [
+                    "https://www.hydroshare.org/resource/",
+                    "http://www.hydroshare.org/resource/",
+                ],
+                "django_irods": "https://www.hydroshare.org/django_irods/download/bags/",
+                "version": "https://www.hydroshare.org/hsapi/resource/{}/scimeta/elements",
+            },
+            "resource": "b8f6eae9d89241cf8b5904033460af61",
+            "version": "1569427757",
+        },
+    )
+]
+
+
+@pytest.mark.parametrize("test_input,expected", test_hosts)
+def test_detect_hydroshare(test_input, expected):
```

Comment:
Could you change this to

```suggestion
def test_detect_hydroshare(test_input):
```

and define `expected` inside the test? I think right now this test only gets called once with these arguments (based on your use of `test_input[0]` and `test_input[1]` etc). However a parametrised test in my mind would run once for each input (once with `test_input[0]` as argument and the expected value for that input, then for `test_input[1]` with expected value for that input, etc).

I think part of my confusion with "what should `fake_urlopen.call_count` be?" comes from the fact that I thought we run this test three times, once for each input. But in reality we run it once.